extra/modules/confiant-ad-quality/src/main/java/org/prebid/server/hooks/modules/com/confiant/adquality/core/BidsMapper.java

```diff
@@ -0,0 +1,38 @@
+package org.prebid.server.hooks.modules.com.confiant.adquality.core;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import org.prebid.server.auction.model.BidderResponse;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.hooks.modules.com.confiant.adquality.model.RedisBidResponseData;
+import org.prebid.server.hooks.modules.com.confiant.adquality.model.RedisBidsData;
+
+import java.util.Collections;
+import java.util.List;
+
+public class BidsMapper {
+
+    public static RedisBidsData bidResponsesToRedisBids(
+            BidRequest bidRequest,
+            List<BidderResponse> bidderResponses) {
+
+        final List<RedisBidResponseData> confiantBidResponses = bidderResponses
+                .stream().map(bidResponse -> RedisBidResponseData
+                        .builder()
+                        .dspId(bidResponse.getBidder())
+                        .bidresponse(BidResponse.builder()
+                                .id(bidRequest.getId())
+                                .cur(bidRequest.getCur().get(0))
+                                .seatbid(Collections.singletonList(SeatBid.builder()
+                                        .bid(bidResponse.getSeatBid().getBids().stream().map(BidderBid::getBid).toList())
+                                        .build()))
```

Comment:
So, all currencies and seatbids except first will be removed after applying this module? I guess that's not what we want. Correct me if I am wrong.

Also, if you really want to rebuild bid responses, please, extract this to separate method, to reduce nesting

Reply:
It will not remove anything. After the scan module can only filter responses if some issues exist. It won't modify existing data.

Unfortunately I have to apply mapping because hook receives BidderResponse model and our API requires BidResponse. And as I understand - BidderResponse contains only single SeatBid. If I've missed something - would appreciate help to find what.

Regarding Currency ... For now that's not very important and honestly I'm not 100% about proper value. We expect to see single currency. So I've used main currency from BidRequest. After investigation of the BidderResponse I see that every BidderBid has currency value. And I'm not sure if they are different which result we expect. 

I'm going to move it into separate method.

Reply:
btw have checked internal Prebid mapping for auctions - https://github.com/prebid/prebid-server-java/blob/master/src/main/java/org/prebid/server/auction/BidResponseCreator.java#L1219
it also takes first currency inside.